Classes/Backend/DynamicFlexForm.php

```diff
@@ -162,6 +162,10 @@ public function parseDataStructureByIdentifierPreProcess(array $identifier)
         } else {
             $dataSourceIdentity = $identifier['tableName'] . '.' . $identifier['fieldName'] . ':' . $identifier['uid'];
         }
+        if(!isset(static::$generatedDataSources[$dataSourceIdentity])) {
```

Comment:
> Maybe this could become a small data storage class like the `TemplatePaths`

I think Christian Kuhn already considered something along those lines but opted against it in favor of just arrays. But I'll mention it to him again - unfortunately though that would require a breaking method signature change at this point.